packages/tests/server/react/infiniteQuery.test.tsx

```diff
@@ -391,4 +392,24 @@ describe('Infinite Query', () => {
       expect(data).toContain('second post');
     }
   });
+
+  test('useInfiniteQuery() is exposed on procedure with optional inputs', () => {
+    const { trpc, appRouter } = factory;
+
+    type AppRouter = typeof appRouter;
+    type Input = inferProcedureInput<AppRouter['paginatedPosts']>;
+
+    type Extends<T, U> = T extends U ? true : false;
+
+    // Optional procedure inputs are unioned with void | undefined.
+
+    assertType<Extends<Input, { cursor?: string }>>(false);
+    assertType<Extends<Input, { cursor?: string } | void>>(true);
+    assertType<Extends<Input, { cursor?: string } | undefined>>(true);
+    assertType<Extends<Input, { cursor?: string } | void | undefined>>(true);
+
+    // Assert 'useInfiniteQuery' is exposed in 'trpc.paginatedPosts'.
+
+    expectTypeOf(trpc.paginatedPosts.useInfiniteQuery).toBeFunction();
+  });
```

Comment:
```suggestion
```



  test('useInfiniteQuery() is **not** exposed if there is not cursor', () => {
    ignoreErrors(async () => {
      // @ts-expect-error 'cursor' is required
      factory.trpc.postById.useInfiniteQuery;
      const ssg = createServerSideHelpers({
        router: factory.appRouter,
        ctx: {},
      });

      // good
      await ssg.paginatedPosts.fetchInfinite({ limit: 1 });

      // @ts-expect-error 'cursor' is required
      await ssg.postById.fetchInfinite({ limit: 1 });
    });
  });